website/blog/unusual-builds-w-bytes.md

```diff
@@ -0,0 +1,224 @@
+   These requests are often annotated with the `prefetcher` action ID and the target ID of either the parent action or the child action.
+
+   We know that in a typical `go_test` rule, there are a few actions involved:
+
+   ```text
+   GoCompilePkg
+   GoCompilePkgExternal
+   => GoLink
+     => TestRunner
+   ```
+
+   and in this case, we can see that all four actions were executed remotely, eliminating the need for any local download.
+
+## The Investigation
+
+While the original intention of the `prefetcher` requests was to [support locally executed actions](https://github.com/bazelbuild/bazel/commit/ea4ad30d4e49b83b62df0e10b5abe2faadea7582), we found that over time, Bazel has been using this action ID for other purposes.
+For example, if a user were to specify a `--remote_download_regex=` pattern to download specific artifacts, those download requests would also be annotated with the `prefetcher` action ID.
+
+As we dug deeper into the issue, we tried to improve the traceability of prefetch downloads in Bazel with [PR#25040](https://github.com/bazelbuild/bazel/pull/25040), which was backported to Bazel 8.1.0 and 7.5.0.
+This change breaks down the `prefetcher` action ID into `input` downloads and `output` downloads, allowing us to narrow down the code paths that were triggering the unwanted downloads.
+In case of input downloads, the target ID would be the consuming/child action's label while in the case of output downloads, the producing/parent action label would be used.
+
+Once we had [upgraded to a Bazel version](./blog/bisect-bazel) with this change, we were able to see that the unwanted downloads were coming from the `output` downloads.
+
+## The Bug
+
+We discovered another code path in Bazel that can trigger unexpected downloads: when a blob's TTL expires.
+
+> So... What is a blob's TTL? 🤔
+
+As an artifact-oriented build system, Bazel relies heavily on artifacts existing in its output path to determine if a build is up-to-date.
+If an artifact exists and matches the expected result, then Bazel will skip the action that produces the artifact.
+However, in a remote build with minimal download, Bazel does not have the artifact locally to validate against and must establish some trust that the artifact was created and stored in the Remote Cache.
+
+Introduced in [PR#17639](https://github.com/bazelbuild/bazel/pull/17639), this "trust" is added to Bazel by storing the expected Time To Live (TTL) of the artifact in the Remote Cache.
+When Bazel executes an action remotely, instead of downloading the outputs referenced inside the ActionResult, it instead just stores the references inside its in-process analysis cache (aka. Skyframe) with an expected TTL that is determined by the flag `--experimental_remote_cache_ttl` (default: 3 hours).
+When the TTL expires, instead of checking if the output artifacts are up-to-date remotely, Bazel downloads the entire artifact to disk.
```

Comment:
We could mention here that this requires the Bazel server to stay alive for the entire TLL, which probably happens more often with BB than elsewhere due to the (usually beneficial) usage of Firecracker etc.

Reply:
I added a few more lines for this. Do give it another read.